examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_pipeline.py

```diff
@@ -0,0 +1,393 @@
+        },
+        "FileToDataFrame": {
+            "module_id": "FileToDataFrame",
+            "module_name": "file_to_dataframe",
+            "namespace": "morpheus_modules",
+            "timestamp_column_name": config.ae.timestamp_column_name,
+            "userid_column_name": config.ae.userid_column_name,
+            "parser_kwargs": {
+                "lines": False, "orient": "records"
+            },
+            "cache_dir": cache_dir,
+            "filter_null": True,
+            "file_type": "JSON",
+            "schema": {
+                "schema_str": source_schema_str, "encoding": encoding
+            }
+        },
+        "DFPSplitUsers": {
+            "module_id": "DFPSplitUsers",
+            "module_name": "dfp_fsplit_users",
+            "namespace": "morpheus_modules",
+            "include_generic": include_generic,
+            "include_individual": include_individual,
+            "skip_users": skip_users,
+            "only_users": only_users,
+            "timestamp_column_name": config.ae.timestamp_column_name,
+            "userid_column_name": config.ae.userid_column_name,
+            "fallback_username": config.ae.fallback_username
+        },
+        "DFPRollingWindow": {
+            "module_id": "DFPRollingWindow",
+            "module_name": "dfp_rolling_window",
+            "namespace": "morpheus_modules",
+            "min_history": 300 if is_training else 1,
+            "min_increment": 300 if is_training else 0,
+            "max_history": "60d" if is_training else "1d",
+            "cache_dir": cache_dir,
+            "timestamp_column_name": config.ae.timestamp_column_name
+        },
+        "DFPPreprocessing": {
+            "module_id": "DFPPreprocessing",
+            "module_name": "dfp_preprocessing",
+            "namespace": "morpheus_modules",
+            "timestamp_column_name": config.ae.timestamp_column_name,
+            "userid_column_name": config.ae.userid_column_name,
+            "schema": {
+                "schema_str": preprocess_schema_str, "encoding": encoding
+            }
+        }
+    }
+
+    # Create a linear pipeline object
+    pipeline = LinearPipeline(config)
+
+    pipeline.set_source(MultiFileSource(config, filenames=list(kwargs["input_file"])))
+
+    # Here we add a wrapped module that implements the full DFPPreprocessing pipeline.
+    pipeline.add_stage(
+        LinearModulesStage(config,
+                           preprocessing_module_config,
+                           input_port_name="input",
+                           output_port_name="output",
+                           output_type=MultiDFPMessage))
+
+    pipeline.add_stage(MonitorStage(config, description="Preprocessing Module rate", smoothing=0.001))
+
+    model_name_formatter = "DFP-azure-{user_id}"
+    experiment_name_formatter = "dfp/azure/training/{reg_model_name}"
+
+    if (is_training):
+
+        # Module configuration
+        training_module_config = {
```

Comment:
I was under the impression that one of the benefits of modules here was you could use a single config to configure the pipeline to run in different ways and that all of the branching would be hidden in the module. @drobison00 is that not true?

Reply:
Discussed with @drobison00. Resolving for now